### ISSUE TITLE
Set performance budget to each URL

### ISSUE THREAD

**Issue by Icecold777** [sitespeedio/sitespeed.io#2139]
Hi there,

In scope of testing integrated in CI/CD process I have to test several URLs generating single HTML report. Those pages has absolutely different page load time making performance budget useless.

I propose to enhance performance budget functionality applying it to each URL specified. In other case default values will be used.
e.g.

```
{  
   "http://example.com":{  
      "browsertime.pageSummary":[  
         {  
            "metric":"statistics.timings.firstPaint.mean",
            "max":300
         },
         {  
            "metric":"statistics.timings.pageTimings.domContentLoadedTime.mean",
            "max":800
         }
      ]
   },
   "default":{  
      "browsertime.pageSummary":[  
         {  
            "metric":"statistics.timings.firstPaint.mean",
            "max":500
         },
         {  
            "metric":"statistics.timings.pageTimings.domContentLoadedTime.mean",
            "max":1500
         }
      ]
   }
}
```
Please let me know if it makes sense or there are any workarounds. 
Best Regards.

**Comment by soulgalore** [comment]
Hi @Icecold777 
if I remember correctly we simplified this and implemented only per run and not per URL ... that maybe was to simplify it too much.

I think the only workaround at the moment is to have multiple budget files and start sitespeed.io per URL.

Best
Peter

**Comment by soulgalore** [comment]
This is implemented in 8.0 checkout https://www.sitespeed.io/documentation/sitespeed.io/performance-budget/#override-per-url
